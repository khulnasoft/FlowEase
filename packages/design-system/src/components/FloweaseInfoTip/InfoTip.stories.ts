import type { StoryFn } from '@storybook/vue3';
import FloweaseInfoTip from './InfoTip.vue';

export default {
	title: 'Atoms/InfoTip',
	component: FloweaseInfoTip,
};

const Template: StoryFn = (args, { argTypes }) => ({
	setup: () => ({ args }),
	props: Object.keys(argTypes),
	components: {
		FloweaseInfoTip,
	},
	template:
		'<flowease-info-tip v-bind="args">Need help doing something? <a href="/docs" target="_blank">Open docs</a></flowease-info-tip>',
});

export const Note = Template.bind({});

export const Tooltip = Template.bind({});
Tooltip.args = {
	type: 'tooltip',
	tooltipPlacement: 'right',
};
