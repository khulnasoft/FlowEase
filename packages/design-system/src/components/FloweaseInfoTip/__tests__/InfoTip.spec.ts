import { render } from '@testing-library/vue';
import FloweaseInfoTip from '../InfoTip.vue';

const slots = {
	default: ['Need help doing something?', '<a href="/docs" target="_blank">Open docs</a>'],
};
const stubs = ['flowease-tooltip'];

describe('FloweaseInfoTip', () => {
	it('should render correctly as note', () => {
		const wrapper = render(FloweaseInfoTip, {
			slots,
			global: {
				stubs,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render correctly as tooltip', () => {
		const wrapper = render(FloweaseInfoTip, {
			slots,
			props: {
				type: 'tooltip',
			},
			global: {
				stubs,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
