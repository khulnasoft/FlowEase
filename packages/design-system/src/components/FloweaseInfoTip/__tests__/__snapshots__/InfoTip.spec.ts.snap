// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FloweaseInfoTip > should render correctly as note 1`] = `"<div class="flowease-info-tip infoTip info note bold"><span class="iconText"><span class="flowease-text compact size-medium regular flowease-icon flowease-icon"><!----></span><span>Need help doing something?<a href="/docs" target="_blank">Open docs</a></span></span></div>"`;

exports[`FloweaseInfoTip > should render correctly as tooltip 1`] = `
"<div class="flowease-info-tip infoTip info tooltip bold">
  <flowease-tooltip-stub popperclass="tooltipPopper" role="tooltip" showafter="0" hideafter="200" autoclose="0" boundariespadding="0" gpuacceleration="true" offset="12" placement="top" popperoptions="[object Object]" strategy="absolute" effect="dark" enterable="true" pure="false" focusonshow="false" trapping="false" stoppoppermouseevent="true" virtualtriggering="false" content="" rawcontent="false" persistent="false" teleported="true" disabled="false" open="false" trigger="hover" triggerkeys="Enter,Space" arrowoffset="5" showarrow="true" justifybuttons="flex-end" buttons="" class=""></flowease-tooltip-stub>
</div>"
`;
